wordCreator.py

```diff
@@ -0,0 +1,294 @@
+            currentMin = 3
+
+        return currentMin
+
+    def setTransitions(self):
+        """
+        Sets the transitions attribute
+        Loops through the text file and finds probabilities that
+        letters transition into other letters
+        Index m, n represents the probability that m transitions to n
+        Index 27 represents a space input
+        """
+        #If using text file
+        if self.txtFile:
+            #Go to beginning of file
+            self.txtFile.seek(0)
+            #Loop through words
+            for line in self.txtFile:
+                line = line.lower()
+                #Increment transition matrix ith letter, i+1th letter by 1
+                for i in range(len(line)-1):
+                    xInd = ord(line[i])
+                    yInd = ord(line[i+1])
+                    if not(xInd > 255 or yInd > 255):
+                        self.transitions[xInd][yInd] += 1
+        #Else use word list
+        if self.words:
+            #Loop through words
+            for i in range(len(self.words)-1):
+                #Increment transition matrix ith letter, i+1th letter by 1
+                xInd = ord(self.words[i])
+                yInd = ord(self.words[i+1])
+                if not(xInd > 255 or yInd > 255):
+                    self.transitions[xInd][yInd] += 1
+        #Normalize Matrix and get totals of each character
+        self.normalizeTransitions()
+
+    def normalizeTransitions(self):
+        """
+        Adds up a row of transitions and divides it by the sum
+        Adds sum to the list of sums (sumList)
+        """
+        #Reinitialize list of character appearances
+        self.sumList = []
+
+        #Loop through each row of Transition Matrix
+        for i in range(len(self.transitions)):
+            #Normalize row and add sum to the list of character appearances
+            self.transitions[i], summation = self.norm(self.transitions[i])
+            self.sumList.append(summation)
+
+    def norm(self, lst):
+        """
+        Receives a list (lst) argument
+        Normalizes the list
+        Returns normalized list and sum of the original entities
+        """
+        #Initialize current sum and make copy of the input list
+        summation = 0
+        tempList = self.copyList(lst)
+
+        summation = self.getSum(lst)
+
+        #Loop through list again, dividing each entry by the sum of all entries
+        for i in range(len(tempList)):
+            tempList[i] = tempList[i]/summation
+
+        #Return normalized list, and the sum of the original values
+        return tempList, summation
+
+    def getSum(self, lst):
+        """
+        Receives list of numbers
+        Returns sum of the list
+        """
+        #Initialize sum
+        summation = 0
+        #Loop through list, adding up values
+        for i in lst:
+            summation += i
+
+        #Return final sum
+        return summation
+
+    def copyList(self, lst):
+        """
```

Comment:
There is a function in python that does this. Use `copy.copy()` or `copy.deepcopy()`. If you are want to know the difference between the two (which is pretty significant difference), you can read the documentation [here](https://docs.python.org/3/library/copy.html).